README.md

```diff
@@ -2,31 +2,114 @@
+1. Install the required dependencies:
+   ```
+   npm install
+   ```
 
-Application data is read at build time. For development purposes, sample data can be imported from `sampleData.ts`.
+1. Generate `index.json`:
+   ```
+   npm run generate-index-json
+   ```
+   The `index.json` is created in the `resources` directory.
 
-#### Building
+1. Copy `index.json` to `./site/public/`.
 
-To build the website, run the following commands:
+### Creating a custom index.json
 
-1. In the root directory, run `npm install`
-2. In the `site` directory, run `npm run build`
+To create an `index.json` of your own, you can either [generate a template](#generating-indexjson) or create a custom file from scratch.
+Your custom JSON file must include the following information:
 
-This will create an optimised static build in the `/site/out` directory.
+* Information about your account or organization (`name` and `description`).
+* Inside the `apps` array, an entry for each app you want to be shown in the index:
 
-### Schemas
+  * `name` must match the application's GitHub repository in your account.
+  * `title` is the human-readable name of the repository.
+  * `description` is the short description of the application.
+  * `manifest` is the alternative name of the west manifest. Defaults to `west.yml`.
+  * `kind` is the type of application
+  * `tags` are the tags that will be used to categorize the application.
+  * `license` is the license type name.
+  * `apps` is the global pattern to find directories containing applications.
+
+Most of the information provided in these entries will be displayed on the application index page.
+For more information about each entry, see `appMetadataSchema` in the `resources/schema.json` file.
+
+#### Schemas
 
 JSON schemas are provided to enforce the shape of the data in the index. They are located in the `resources` directory. To edit the schemas, make any changes in `site/src/schema.ts` and run `npm run generate-schemas` from the root directory.
+
+### Serving a local server
+
+Once you have your `index.json` file, you can set up a local server to test it:
+
+1. Go to the `site` directory.
+
+1. Install the required dependencies:
+   ```
+   npm install
+   ```
+
+1. Build the website locally:
+   ```
+   npm run build
+   ```
+   This will create a static build in the `/site/out` directory.
+
+### Verifying website locally
+
+This repository includes a static website that displays the contributed applications in a searchable frontend. It's developed using the [Next.js](https://nextjs.org/) React framework, and uses [Tailwind](https://tailwindcss.com/) for styling.
+
+To verify your application index website locally:
+
+1. Start the website locally:
+   ```
+   npm run dev
+   ```
+
+1. Access the site in the browser by going to `localhost:3000/ncs-app-index`.
+
+1. Check if the `index.json` is available by going to `localhost:3000/ncs-app-index/index.json`.
```

Comment:
I'd be simpler just to click rather than copy-paste the addess, does this work?
```suggestion
1. Access the site in the browser by going to [localhost:3000/ncs-app-index](http://localhost:3000/ncs-app-index).

1. Check if the `index.json` is available by going to [localhost:3000/ncs-app-index/index.json](http://localhost:3000/ncs-app-index/index.json).
```